pkg/apis/spinnaker/v1alpha2/spinnakerservice_types.go

```diff
@@ -124,42 +151,47 @@ type SpinnakerServiceStatus struct {
 	AccountCount int `json:"accountCount,omitempty"`
 }
 
-// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
-
-// SpinnakerService is the Schema for the spinnakerservices API
-// +k8s:openapi-gen=true
-// +kubebuilder:subresource:status
-// +kubebuilder:printcolumn:name="version",type="string",JSONPath=".status.version",description="Version"
-// +kubebuilder:printcolumn:name="lastConfigured",type="date",JSONPath=".status.lastConfigurationTime",description="Last Configured"
-// +kubebuilder:printcolumn:name="status",type="string",JSONPath=".status.status",description="Status"
-// +kubebuilder:printcolumn:name="services",type="number",JSONPath=".status.serviceCount",description="Services"
-// +kubebuilder:printcolumn:name="url",type="string",JSONPath=".status.uiUrl",description="URL"
-// +kubebuilder:printcolumn:name="apiUrl",type="string",JSONPath=".status.apiUrl",description="API URL",priority=1
-// +kubebuilder:resource:path=spinnakerservices,shortName=spinsvc
-type SpinnakerService struct {
-	metav1.TypeMeta   `json:",inline"`
-	metav1.ObjectMeta `json:"metadata,omitempty"`
-
-	Spec   SpinnakerServiceSpec   `json:"spec,omitempty"`
-	Status SpinnakerServiceStatus `json:"status,omitempty"`
+// GetUpdateHash updates the hash at key `key` and returns the prior copy if one existed
+// LastDeployed should then contain the hash and the time if updateTime is true or if there was no hash
+func (s *SpinnakerServiceStatus) UpdateHashIfNotExist(key, hash string, t time.Time, updateTime bool) HashStatus {
```

Comment:
I think we could move these functions (UpdateHashIfNotExist, GetValidationSettings and NeedsValidation) to some other file, to keep this "types" file only concerned in defining types for the CRD.